pkg/storage/stores/shipper/compactor/deletion/delete_requests_manager.go

```diff
@@ -203,13 +203,15 @@ func (d *DeleteRequestsManager) MarkPhaseFinished() {
 	}
 }
 
-func (d *DeleteRequestsManager) IntervalMayHaveExpiredChunks(_ model.Interval, userID string) bool {
+func (d *DeleteRequestsManager) IntervalMayHaveExpiredChunks(interval model.Interval, userID string) bool {
 	d.deleteRequestsToProcessMtx.Lock()
 	defer d.deleteRequestsToProcessMtx.Unlock()
 
 	if userID != "" {
 		for _, deleteRequest := range d.deleteRequestsToProcess {
-			if deleteRequest.UserID == userID {
+			if deleteRequest.UserID == userID &&
+				deleteRequest.StartTime <= interval.End &&
```

Comment:
In your comment on the PR you mention if an interval overlaps a delete request, but this logic requires the interval to be fully within the delete request time period (so excluding an overlap at the start or end of the interval). Is that intended? Would there be a situation where we'd want an `||` here instead?

Reply:
This actually handles overlaps because any overlapping interval will always start before the end of the delete and end after the start of the delete. Fully overlapping is just a special case. I've  included a picture

![Untitled-2022-05-25-1713](https://user-images.githubusercontent.com/1413241/171750326-e8abffa5-a74d-4cbf-bec4-cad16b064f4b.png)